### PR TITLE
Multiply blending

### PR DESCRIPTION
**Pull request by christopherbauer**
This is a branch that satisfies some base criteria for multiply blending per issue https://github.com/JimBobSquarePants/ImageProcessor/issues/294. Discussion re: the implementation of multiply blending can continue here as work moves forward while generic information about blending can occur in the issue.

Or if this is okay as is we can merge in and investigate other modes.

### CODE REVIEW

**Comment by christopherbauer** [comment]
RE: The spec to black/white: multiplying by black would be multiplying R G and B with 0 which would return black anyway, so this is more of a 'skip the cpu cycles if it is black, we don't really need to bother. Same thing for white - multiplying by 1f across the board results in the same value as you started. Though I suppose the implementation of equals in the lib will incur more cycles then just performing the multiplication outright, huh? :)

**Comment by JimBobSquarePants** [comment]
I've not tested that actually but I would imagine it would do since I don't actually use as strict equals in my equality operators, rather compare to an epsilon.

**Comment by JimBobSquarePants** [comment]
This is my fault for giving you a name that already exists but could you separate out the colorspace transforms into a separate file, they make Color.cs difficult to traverse. Call it ColorSpacialTransforms

**Comment by christopherbauer** [comment]
Hey Jim,

That was just a weird Github merge error in the diffs. I don't know why it appears to be moving Color out of the file ColorTransform.cs but rest assured your sanity is in check, the file did not exist prior to the branch :)

**Comment by JimBobSquarePants** [comment]
Nah something's wrong here chum. A quick dig in your repo yields 

https://github.com/christopherbauer/ImageProcessor/blob/MultiplyBlending/src/ImageProcessor/Colors/Color.cs

With the colorspace translations present. In my V3 branch you will find a file ColorTransforms.cs that contains those methods.

**Comment by christopherbauer** [comment]
OH! It's a file but the file contains a public partial Color declaration. I'm used to files being named after the classes that contain them and completely missed it. I must have moved the multiply declaration into ColorTransform.cs then alt+entered the partial color struct declaration. I'll clean up the commit ASAP.

**Comment by JimBobSquarePants** [comment]
No worries :smile: Yeah, taking a slightly different approach with that struct. There's far too much to go in one file.

**Comment by christopherbauer** [comment]
I just squashed my commit tree down to a single commit for brevity's sake and to keep from polluting the branch with my bad commits :+1:

File: src/ImageProcessor/Colors/ColorSpacialTransforms.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿// <copyright file="ColorSpacialTransforms.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageProcessor
+{
+    public struct ColorSpacialTransforms
```
Comment (JimBobSquarePants) [inline src/ImageProcessor/Colors/ColorSpacialTransforms.cs]: Should be `public partial struc Color`

File: src/ImageProcessor/Colors/ColorSpacialTransforms.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            if (destination == Color.Black)
+            {
+                return Color.Black;
+            }
+            if (destination == Color.White)
+            {
+                return source;
+            }
+            return new Color(
+                (source.R * destination.R).Clamp(0f, 1f),
```
Comment (JimBobSquarePants) [inline src/ImageProcessor/Colors/ColorSpacialTransforms.cs]: Once you use the correct partial you can then multiply the backing vectors. Don't limit as I want to support HDR

File: src/ImageProcessor/Colors/ColorSpacialTransforms.cs (hunk 1 above)
Comment (christopherbauer) [inline src/ImageProcessor/Colors/ColorSpacialTransforms.cs]: I'm falling to pieces!

Comment (JimBobSquarePants) [inline src/ImageProcessor/Colors/ColorSpacialTransforms.cs]: Tell me about it! I'm banging my head against a brick wall just now trying to speed up the resampler.

**Comment by christopherbauer** [comment]
Okay - working with the backing vector, corrected the struct issue, merged from origin, and squashed the commits so it's not messy!

**Comment by JimBobSquarePants** [comment]
perfick

**Pull request merged**